packages/ckeditor5-link/tests/autolink.js

```diff
@@ -110,6 +110,56 @@ describe( 'AutoLink', () => {
 			);
 		} );
 
+		it( 'does not add linkHref attribute on enter when the link is selected', () => {
+			setData( model, '<paragraph>[https://www.cksource.com]</paragraph>' );
+
+			editor.execute( 'enter' );
+
+			expect( getData( model ) ).to.equal(
+				'<paragraph>[]</paragraph>'
+			);
+		} );
+
+		it( 'does not add linkHref attribute on enter when the whole paragraph containing the link is selected', () => {
+			setData( model, '<paragraph>[This feature also works with e-mail addresses: https://www.cksource.com]</paragraph>' );
+
+			editor.execute( 'enter' );
+
+			expect( getData( model ) ).to.equal(
+				'<paragraph>[]</paragraph>'
+			);
+		} );
+
+		it( 'adds linkHref attribute on enter when the link (that contains www) is partially selected (end)', () => {
+			setData( model, '<paragraph>https://www.ckso[urce.com]</paragraph>' );
+
+			editor.execute( 'enter' );
+
+			expect( getData( model ) ).to.equal(
+				'<paragraph><$text linkHref="https://www.ckso">https://www.ckso</$text></paragraph><paragraph>[]</paragraph>'
+			);
+		} );
```

Comment:
Are you sure this is OK? I mean:

*   Why should the `www` prefix change the auto-link behavior? It does now. For instance, starting with `<paragraph>https://www.ckso[urce.com]</paragraph>` vs. `<paragraph>https://ckso[urce.com]</paragraph>` ends with a link (former) and no link (latter).
*   `<paragraph>https://www.ckso[urce.com]</paragraph>`  also ends up as a broken link: `linkHref="https://www.ckso"`. This is not a valid link (it's just a fragment of the original link) and from the user point of view, this is a bug.
*   It looks like something is wrong with `URL_REG_EXP`. I don't think we should ever auto-link if a link is partially selected in the first place.

If you agree with me, can you please create a follow-up issue @panr? The test can stay, it's harmless. Maybe a comment that this is something weird would do the trick for now.

Reply:
It's probably because the `URL_REG_EXP` considers `ckso` a domain:

![](https://user-images.githubusercontent.com/1099479/92606407-2e6a2f00-f2b3-11ea-8576-79ed2d30f86e.png)

![](https://user-images.githubusercontent.com/1099479/92605993-b7349b00-f2b2-11ea-939d-b0601264149f.png)

Reply:
cc @jodator

Reply:
@oleq I agree that this incorrect behavior. I was surprised as well at first that this didn't pass the first version of the test, but I thought that this is how should it be, knowing how we handle validation ;-D

I'll create a followup to this issue 👍